Topics/Development_Process/Serverless_AWS_Lambda.md

```diff
@@ -0,0 +1,69 @@
+# Serverless Computing with AWS Lambda
+
+## Table of Contents
+
+### [Introduction](#introduction-1)
+
+### [Benefits of Serverless Computing](#benefits-of-serverless-computing-1)
+
+### [Disadvantages of Serverless Computing](#disadvantages-of-serverless-computing-1)
+
+### [AWS Lambda](#aws-lambda-1)
+
+#### [Creating Lambda Functions in the AWS Console](#creating-lambda-functions-in-the-aws-console-1)
+
+#### [Invoking Lambda Functions from the AWS Console](#invoking-lambda-functions-from-the-aws-console-1)
+
+### [Additional Resources](#additional-resources-1)
+
+## Introduction
+
+It can be hard to set up and maintain your backend services. There are a ton of hosting options to choose from, and they can get pretty pricey.
+
+Serverless computing allows you to to build and run applications without needing to manage servers. Cloud providers such as AWS, Azure, Google Cloud automatically provision servers based on demand.
+
+## Benefits of Serverless Computing
+
+As a developer, using serverless computing for your backend services has many benefits.
+
+1. **Lower cost**: With serverless computing, you only pay for what you use. Traditional cloud computing typically requires you to pay for a fixed server, which often results in you having to pay for idle time and unused space.
+2. **Scalability**: Because it is not up to the developer to allocate and maintain servers, developers do not have to worry about scaling servers up or down, as it is up to the cloud provider to do so.
+3. **Easier maintenance**: The developer/user does not need to manage their backend services, as it is up to the cloud provider to maintain the servers. With containers or other deployment options, developers are responsible for updating and maintaining them.
+4. **Deployment time**: Serverless functions finish deploying as soon as their code is updated, making them generally faster than containers, or virtual machines.
+
+## Disadvantages of Serverless Computing
+
+There are some situations where you might not want to use serverless computing.
```

Comment:
I feel like you could expand more on this. There might be more situations where serverless isn't worth it, such as in cases where you want more control over your program configs, etc

Reply:
Do you mean being able to choose the type or environment of the server (because the cloud provider does that for you)?

Reply:
Hi Elsie, sort of. There are some things that AWS Lambda does not give you control over, such as the type of OS you use (some systems may require Ubuntu to work, or Debian, etc.). There may be other things that you can't change that I'm also missing but might be worth mentioning as another drawback

Reply:
I believe most serverless compute services allow you to choose higher level options such as the type of operating system, but not lower level environment options. I will definitely add this to the article!

Reply:
Awesome sounds great! It'll be good for people who are trying to build specialized tools that require a very very specific environment :)